### ISSUE TITLE
Compilation broken on Ruby 3.4 (preview2 and rc1)

### ISSUE THREAD

**Issue by larouxn** [ankane/ruby-polars#94]
It _appears_ that compilation of `polars-df` on Ruby 3.4 (tried `preview2` and `rc1`, full release coming in 5 days) is broken. Running a `bundle install` fails while trying to compile native extensions for `polars-df` both on a standard x86 Linux box and on macOS 15.2 + Apple M3 Pro i.e. ARM. I also tried bumping `rb-sys` from `0.9.102` to `0.9.103` just in case.

```console
┃┃    Compiling rb-sys v0.9.102
┃┃ error: failed to run custom build command for `rb-sys v0.9.102`
┃┃ Caused by:
┃┃ process didn't exit successfully:
┃┃ `/Users/larouxn/.gem/ruby/3.4.0-rc1/gems/polars-df-0.15.0/ext/polars/target/release/build/rb-sys-713177211409f799/build-script-main` (exit status: 101)
┃┃   --- stdout
┃┃   cargo:rerun-if-env-changed=RUBY
```

[full console log as a Gist](https://gist.github.com/larouxn/368df2601fdc21e608d9d3e893aa9971) (due to character limit)

If there's anything I can do to help debug this and/or contribute to a fix, please let me know. Thanks!

**Comment by ankane** [comment]
Hi @larouxn, it’ll be supported once 3.4.0 is released.

**Comment by ksr-13** [comment]
This is still broken on release ruby v 3.4.1

**Comment by larouxn** [comment]
> This is still broken on release ruby v 3.4.1

Are you trying on `master`? I believe it's working now but a release has not been cut yet. Last release was November 20th.

https://github.com/ankane/ruby-polars/releases/tag/v0.15.0

**Comment by ankane** [comment]
Just released 0.16.0, which supports Ruby 3.4.

**Comment by ksr-13** [comment]
Thank you @ankane. Its working.
